src/components/SignUpForm/index.test.js

```diff
@@ -0,0 +1,409 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+import SignUpForm from "./index";
+
+describe("SignUpForm", () => {
+  it("renders correctly and matches the snapshot", () => {
+    const tree = renderer.create(<SignUpForm />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  const mockSubmit = jest.fn(
+    (firstName, lastName, email, password, confirmPassword, acceptTerms) => {
+      return Promise.resolve({
+        firstName,
+        lastName,
+        email,
+        password,
+        confirmPassword,
+        acceptTerms,
+      });
+    }
+  );
+
+  beforeEach(() => {
+    render(<SignUpForm submit={mockSubmit} />);
+  });
+
+  it("should display required error when value is empty", async () => {
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(5);
+    expect(mockSubmit).not.toBeCalled();
+  });
+
+  it("should display matching error when email is invalid", async () => {
+    fireEvent.input(screen.getByRole("textbox", { name: /firstName/i }), {
+      target: {
+        value: "name",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /lastName/i }), {
+      target: {
+        value: "surname",
+      },
+    });
+
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/^password/i), {
+      target: {
+        value: "password",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/repeat password/i), {
+      target: {
+        value: "password",
+      },
+    });
+
+    fireEvent.click(screen.getByRole("checkbox"));
+
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(1);
+    expect(mockSubmit).not.toBeCalled();
+    expect(screen.getByRole("textbox", { name: /firstName/i }).value).toBe(
+      "name"
+    );
+    expect(screen.getByRole("textbox", { name: /lastName/i }).value).toBe(
+      "surname"
+    );
+    expect(screen.getByRole("textbox", { name: /email/i }).value).toBe("test");
+    expect(screen.getByPlaceholderText(/^password/i).value).toBe("password");
+    expect(screen.getByPlaceholderText(/repeat password/i).value).toBe(
+      "password"
+    );
+    expect(screen.getByRole("checkbox").checked).toBe(true);
+  });
+
+  it("should display matching error when email domain is invalid", async () => {
```

Comment:
delete this feature. in genera you don't want to make assumptions about users emails, names, etc. you will block users randomly as a result

Reply:
file name should be index.test.jsx for consistency @Jamil-Bailony

Reply:
we talked about it, the test file is a JavaScript file not a JSX file @jennethydyrova

Reply:
I don't remember that we talked about it but whole team is using jsx if you check components

Reply:
but I guess you used js in all your tests

Reply:
actually its jsx. because you are using jsx

Reply:
```
    render(<SignUpForm submit={mockSubmit} />);
```

code like this is not allowed in normal js. it will also not highlight properly in .js. therefore its jsx